lib/ffi-icu/number_formatting.rb

```diff
@@ -68,17 +68,22 @@ def format(number)
             case number
             when Float
               needed_length = Lib.unum_format_double(@f, number, out_ptr, needed_length, nil, error)
-            when Fixnum
-              needed_length = Lib.unum_format_int32(@f, number, out_ptr, needed_length, nil, error)
+            when Integer
+              begin
+                # Try doing it fast, for integers that can be marshaled into an int64_t
+                needed_length = Lib.unum_format_int64(@f, number, out_ptr, needed_length, nil, error)
+              rescue RangeError
+                # Fall back to stringifying in Ruby and passing that to ICU
+                string_version = number.to_s
+                needed_length = Lib.unum_format_decimal(@f, string_version, string_version.bytesize, out_ptr, needed_length, nil, error)
```

Comment:
I believe we'll need to handle the case where `#unum_format_decimal` isn't defined - as it may not always be supported https://github.com/erickguan/ffi-icu/blob/0c6f8b201f3994fa10c741ff5dbb146cb83f4a9d/lib/ffi-icu/lib.rb#L418-L421

Though it looks to have been a part of ICU [for quite a while now](https://unicode-org.github.io/icu-docs/apidoc/dev/icu4c/unum_8h.html#a480ae7530a43cbdc9c3a33a726198c72), in which case we could clean up that assignment / some conditional checks in a followup.

Reply:
I'm not entirely sure what we should _do_ if unum_format_decimal is not defined though - I don't think there's another way to format a number that won't fit in an int64_t.

Reply:
I guess the only other thing we can do is raise a more descriptive error:

```
raise RangeError, "Number #{number} is too big to fit in int64_t and your ICU version is too old to have unum_format_decimal" unless defined? Lib.unum_format_decimal
```

Reply:
>I guess the only other thing we can do is raise a more descriptive error

Seems reasonable - thoughts @damiann ?

Reply:
@codealchemy I went ahead and added that check